File: Backend/src/main/java/pl/prg/ba/Model/Task.java
package pl.prg.ba.Model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import pl.prg.ba.Enums.Taks.TaskStatus;
@Entity
@Table(name = "tasks")
@Getter
@Setter

public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "varchar(255)")
    private TaskStatus taskStatus;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: Frontend/src/app/Model/Task/task.ts
import {UserList} from "../UserList/user-list";
import {ListCategory} from "../ListCategory/list-category";

export class Task {
  id: number;
  name: string;
  taskStatus: string;
  listCategory: ListCategory;
  constructor() {
    this.id = 0;
    this.name = '';
    this.taskStatus = '';
    this.listCategory = new ListCategory();
  }
}
